lib/tokenFactory.js

```diff
@@ -27,19 +28,45 @@ class TokenFactory extends BaseFactory {
     }
 
     /**
-     * Create a token model
+     * Create a token model, returning the model including the unhashed value
      * @method create
      * @param  {Object}     config
      * @param  {String}     config.userId            The ID of the associated user
-     * @param  {String}     config.value             The token value
      * @param  {String}     config.name              The token name
      * @param  {String}     config.description       The token description
      * @return {Promise}
      */
     create(config) {
-        config.lastUsed = null;
+        let value;
 
-        return super.create(config);
+        return generateToken.generateValue()
+            .then((bytes) => {
+                value = bytes;
+                config.hash = generateToken.hashValue(bytes);
+                config.lastUsed = null;
+
+                return super.create(config);
+            }).then((model) => {
+                model.value = value;
+
+                return model;
+            });
+    }
+
+    /**
+     * Get a token
+     * @method get
+     * @param  {Mixed}     config
+     * @param  {String}    [config.value]    (Un-hashed) value of token to look for
+     * @return {Promise}
+     */
+    get(config) {
+        if (config && config.value) {
+            config.hash = generateToken.hashValue(config.value);
+            delete config.value;
+        }
+
+        return super.get(config);
```

Comment:
You mean like delete the `hash` field before returning it?

We can't do that in the model layer, it doesn't let you delete the fields that go in the datastore (unless we want to mark the field as one that can be deleted when creating the mdoel, but I'm not sure what effect that would have if you tried to call save)

For secrets it looks like the value is deleted in the API layer before they're returned.